File: src/main/java/com/webforj/samples/components/SplitterBox.java
package com.webforj.samples.components;

import com.webforj.bundle.annotation.BundleEntry;
import com.webforj.component.Composite;
import com.webforj.component.layout.flexlayout.FlexAlignment;
import com.webforj.component.layout.flexlayout.FlexJustifyContent;
import com.webforj.component.layout.flexlayout.FlexLayout;

@BundleEntry("css/splitter/splitter-box.css")
public class SplitterBox extends Composite<FlexLayout> {
  public enum Theme {
    INFO,
    SUCCESS,
    WARNING,
    PRIMARY
  }

  public SplitterBox(String label, Theme type) {
    super();
    FlexLayout layout = getBoundComponent();
    layout.setText(label);
    layout.setJustifyContent(FlexJustifyContent.CENTER);
    layout.setAlignment(FlexAlignment.CENTER);
    layout.addClassName("splitter-box", "splitter-box--" + type.name().toLowerCase());
  }
}

File: src/main/kotlin/com/webforj/samples/views/splitter/SplitterBox.kt
package com.webforj.samples.views.splitter

import com.webforj.concern.HasComponents
import com.webforj.kotlin.dsl.WebforjDsl
import com.webforj.kotlin.dsl.init
import com.webforj.samples.components.SplitterBox

@WebforjDsl
fun @WebforjDsl HasComponents.splitterBox(
  label: String,
  theme: SplitterBox.Theme,
  block: @WebforjDsl SplitterBox.() -> Unit = {}): SplitterBox {
  return init(SplitterBox(label, theme), block)
}
